src/cephadm/cephadm.py

```diff
@@ -1246,7 +1248,11 @@ def deploy_daemon_units(
         call_throws(ctx, ['systemctl', 'enable', unit_name])
     if start:
         clean_cgroup(ctx, ident.fsid, unit_name)
-        call_throws(ctx, ['systemctl', 'start', unit_name])
+        try:
+            call_throws(ctx, ['systemctl', 'start', unit_name])
+        except Exception as e:
+            logger.error(f'systemctl start failed for {unit_name}: {str(e)}')
+            raise DaemonStartException()
```

Comment:
could we use the exception msg `e` when to create `DaemonStartException`?

Reply:
I don't think it's necessary. The mgr will never see the exception contents since we handle this by just doing a `sys.exit()` with a specific rc. The only difference would maybe be something added to a traceback in the logs, but I think that's not needed since we're logging the issue right before we raise the exception.